cardano-config/README.md

```diff
@@ -1,3 +1,18 @@
 # Cardano Configuration
 
-This package exposes the types and functions necessary to configure the `cardano-node`.
+This package exposes functions to provide git information for `cardano-node`.
+
+`cardano-node` support building via `nix` and `cabal`
+
+When building with `nix` the git executable and git metadata isn't available so the
+git revision is embedded as a series of 40 zeros during the build. After the nix build
+is finished the executable is patched with the correct git sha. See [set-git-rev.hs][set-git-rev.hs]
+
+For `cabal` the Template Haskell function `gitRev` executes in the context of the current git checkout,
+requiring the `git` executable and git metadata is available at compile time. This fails on cross-compiling
+ARM executables, for cross-compiling use nix.
```

Comment:
I don't follow.

Reply:
We know this doesn't work with `nix`, though I still think we should be able to improve this with nix, and yes cross compilation is exclusive through nix, so this can't work,. It's in the `cabal` section though, so I'm fairly confused.

Reply:
Perhaps my wording isn't descriptive enough here and I can drop the details about cross-compiling.
The short version is `cabal` builds require `git` and a git checkout while building to get an accurate git sha. 

It took me a while to discover how both nix and cabal, get the right git sha patched into the final executable, I wanted to capture that knowledge somewhere. 

https://github.com/input-output-hk/cardano-node/issues/3440